### DISCUSSION TITLE
Any way to access dock labels from apps?

### DISCUSSION THREAD

**Discussion by carlos-gtz** [Hammerspoon/hammerspoon#3251] [Q&A] [answered]
I would like to add a menu with slack notifications unread 🤔 The easiest would be to do some API access through REST... but if we already have the notification in the dock app... then that would be better as I wouldn't need to access Slack API and polling to get the notifications.

Is there a way to access the Dock and read the label on an app (slack app) to use it in any cool way?

**Comment by latenitefilms** [comment, marked as answer]
You can read the `AXStatusLabel` of the dock item via the Accessibility API.

See: http://www.hammerspoon.org/docs/hs.axuielement.html

**Reply by carlos-gtz** [reply]
Thank you 🙏

**Reply by carlos-gtz** [reply]
I came up with this code:

```lua
Dock = hs.axuielement.applicationElement('Dock'):attributeValue('AXChildren')

for _, v in pairs(Dock[1].AXChildren) do
  if v:attributeValue('AXTitle') == 'Slack' then
    local label = v:attributeValue('AXStatusLabel') or '0'
    print(label)
  end
end
```

Thank you for your help.

**Reply by latenitefilms** [reply]
You could probably add a `break` after the print to escape the `for` loop too.
